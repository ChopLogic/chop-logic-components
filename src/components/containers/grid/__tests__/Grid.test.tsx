import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import ChopLogicGrid from '../Grid';

describe('ChopLogicGrid', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<ChopLogicGrid columns={[]} data={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
