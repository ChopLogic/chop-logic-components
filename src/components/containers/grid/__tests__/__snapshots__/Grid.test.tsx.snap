// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicGrid > should match the snapshot 1`] = `
<DocumentFragment>
  <table
    class="sc-braxZu hgBksX"
  >
    <colgroup>
      <col
        class="sc-iCZwEW hZKiWH"
      />
      <col
        class="sc-iCZwEW gywXzI"
      />
      <col
        class="sc-iCZwEW gywXzI"
      />
      <col
        class="sc-iCZwEW gywXzI"
      />
    </colgroup>
    <thead
      class="sc-gJhJTp cboyra"
    >
      <tr
        class="sc-dsLQwm jqsKNm"
      >
        <th
          class="sc-fQpRED ezRTCh"
        >
          <div
            class="sc-knesRu eVQQCM"
          >
            <input
              class="sc-dnaUSb cVcZeW"
              id="select_all_:r0:"
              type="checkbox"
            />
            <label
              class="sc-kkmypM gdLSUN"
              for="select_all_:r0:"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </th>
        <th
          class="sc-fQpRED ezRTCh"
        >
          Company
        </th>
        <th
          class="sc-fQpRED ezRTCh"
        >
          Contact
        </th>
        <th
          class="sc-fQpRED ezRTCh"
        >
          Country
        </th>
        <th
          class="sc-fQpRED ezRTCh"
        >
          Phone
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-elDIKY jigrUQ"
    >
      <tr
        class="sc-la-DxNn kXFzaW"
      >
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          <div
            class="sc-knesRu eVQQCM"
          >
            <input
              class="sc-dnaUSb cVcZeW"
              id="select_row_row1"
              type="checkbox"
            />
            <label
              class="sc-kkmypM gdLSUN"
              for="select_row_row1"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        />
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Maria
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Germany
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          123-345-7890
        </td>
      </tr>
      <tr
        class="sc-la-DxNn kXFzaW"
      >
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          <div
            class="sc-knesRu eVQQCM"
          >
            <input
              class="sc-dnaUSb cVcZeW"
              id="select_row_row2"
              type="checkbox"
            />
            <label
              class="sc-kkmypM gdLSUN"
              for="select_row_row2"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        />
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Francisco
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Mexico
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          123-345-7890
        </td>
      </tr>
      <tr
        class="sc-la-DxNn kXFzaW"
      >
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          <div
            class="sc-knesRu eVQQCM"
          >
            <input
              class="sc-dnaUSb cVcZeW"
              id="select_row_row3"
              type="checkbox"
            />
            <label
              class="sc-kkmypM gdLSUN"
              for="select_row_row3"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Ernst Handel
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Roland Mendel
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Austria
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          123-345-7890
        </td>
      </tr>
      <tr
        class="sc-la-DxNn kXFzaW"
      >
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          <div
            class="sc-knesRu eVQQCM"
          >
            <input
              class="sc-dnaUSb cVcZeW"
              id="select_row_row4"
              type="checkbox"
            />
            <label
              class="sc-kkmypM gdLSUN"
              for="select_row_row4"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Island Trading
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Helen Bennett
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          UK
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          123-345-7890
        </td>
      </tr>
      <tr
        class="sc-la-DxNn kXFzaW"
      >
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          <div
            class="sc-knesRu eVQQCM"
          >
            <input
              class="sc-dnaUSb cVcZeW"
              id="select_row_row5"
              type="checkbox"
            />
            <label
              class="sc-kkmypM gdLSUN"
              for="select_row_row5"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Laughing Bacchus
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        />
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Canada
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          123-345-7890
        </td>
      </tr>
      <tr
        class="sc-la-DxNn kXFzaW"
      >
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          <div
            class="sc-knesRu eVQQCM"
          >
            <input
              class="sc-dnaUSb cVcZeW"
              id="select_row_row6"
              type="checkbox"
            />
            <label
              class="sc-kkmypM gdLSUN"
              for="select_row_row6"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        />
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Giovanni
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          Italy
        </td>
        <td
          class="sc-iKTcqh hQfLEQ"
        >
          123-345-7890
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
